lib/Dialect/OM/OMOps.cpp

```diff
@@ -415,25 +424,15 @@ void circt::om::ClassOp::addNewFieldsOp(mlir::OpBuilder &builder,
   }
   // Also store the locations incase there's some other analysis that might
   // be able to use the default FusedLoc representation.
-  builder.create<ClassFieldsOp>(
-      builder.getFusedLoc(locs, builder.getArrayAttr(locAttrs)), values);
+  builder.create<ClassFieldsOp>(builder.getFusedLoc(locs), values,
+                                builder.getArrayAttr(locAttrs));
 }
 
 mlir::Location circt::om::ClassOp::getFieldLocByIndex(size_t i) {
-  Location loc = this->getFieldsOp()->getLoc();
-  if (auto locs = dyn_cast<FusedLoc>(loc)) {
-    // Because it's possible for a user to construct a fields op directly and
-    // place a FusedLoc that doersn't follow the storage format of
-    // addNewFieldsOp, we assert the information has been stored appropriately
-    ArrayAttr metadataArr = dyn_cast<ArrayAttr>(locs.getMetadata());
-    assert(metadataArr && "Expected fused loc to store metadata array");
-    assert(i < metadataArr.size() &&
-           "expected index to be less than array size");
-    LocationAttr locAttr = dyn_cast<LocationAttr>(metadataArr[i]);
-    assert(locAttr && "expected metadataArr entry to be location attribute");
-    loc = Location(locAttr);
-  }
-  return loc;
+  auto fieldLocs = this->getFieldsOp().getFieldsLocs();
+  if (!fieldLocs)
```

Comment:
Is there a valid case where this can be None, or is this just being flexible? If I'm understanding correctly, this should be set for any valid user input (at least from FIRRTL), since there's really only one path into this IR. I guess what I'm getting at is if this should be an assertion that fieldLocs is not null, and has exactly the number of elements as there are fields, since that is the invariant we already expect the compiler implementation to maintain.

Reply:
We could maintain that assertion internally to the compiler, but it could be possible for some input `.mlir`, e.g. writing tests directly in the `om` dialect, to omit the optional attr.  If we implement a custom parser that fills that the attribute and make it non optional, then we could change this to an assertion/assumption.

Reply:
Ok fair enough, I think it's up to you. IMHO it would be fine to say this is required, if we want it to be, and tests would have to specify it too. On the other hand, it really is optional information, and leaving it as such makes sense to me. Personally I'd prefer to keep things simple, but if we're adding a custom parser/printer to gate this behind the "should print debug info" flag, then I guess we could have it be required and let the parser fill in unknowns in the case we're taking in .mlir text in tests, etc.